src/unix/bsd/freebsdlike/dragonfly/mod.rs

```diff
@@ -115,4 +115,12 @@ extern {
     pub fn mprotect(addr: *mut ::c_void, len: ::size_t, prot: ::c_int)
                     -> ::c_int;
     pub fn clock_gettime(clk_id: ::c_ulong, tp: *mut ::timespec) -> ::c_int;
+
+    pub fn pthread_attr_get_np(tid: ::pthread_t,
```

Comment:
yes. https://www.freebsd.org/cgi/man.cgi?query=pthread_attr_get_np. Also, guard page code for FreeBSD seems to be missing in Rust. Did you run "gmake check" when building rust? One test that overflows the stack on purpose should also fail for FreeBSD (it fails on DragonFly). I will add those functions then to the common code base for both DragonFly and FreeBSD